components/style/values/specified/position.rs

```diff
@@ -608,6 +611,30 @@ impl Parse for TemplateAreas {
 
 trivial_to_computed_value!(TemplateAreas);
 
+#[derive(Clone, Debug, PartialEq, ToCss)]
+/// Arc type for `Arc<TemplateAreas>`
+pub struct TemplateAreasArc(pub Arc<TemplateAreas>);
+
+impl Parse for TemplateAreasArc {
+    fn parse<'i, 't>(
+        context: &ParserContext,
+        input: &mut Parser<'i, 't>,
+    ) -> Result<Self, ParseError<'i>> {
+        let parsed = TemplateAreas::parse(context, input)?;
+
+        Ok(TemplateAreasArc(Arc::new(parsed)))
+    }
+}
+
+#[cfg(feature = "gecko")]
+impl MallocSizeOf for TemplateAreasArc {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        self.0.conditional_size_of(ops)
```

Comment:
This isn't quite right. `TemplateAreasArc` is basically the same as `Arc`, so it should implement the same traits: one or more of `MallocUnconditionalShallowSizeOf`, `MallocUnconditionalSizeOf`, `MallocConditionalShallowSizeOf`, `MallocConditionalSizeOf`. It should *not* implement `MallocSizeOf` or `MallocShallowSizeOf`. See components/malloc_size_of/lib.rs for more details.

Having said that: does `TemplateAreasArc` need to be a newtype? Could it just be a typedef for `Arc<TemplateAreas>`? Maybe you need a newtype so you can define additional  methods?

TemplateAreasArc

Reply:
Right, it needs to be a newtype so it can implement `ToCss` and such.

Reply:
Though arguably that should be generically implemented... `ToComputedValue` is trickier though.

Reply:
@nnethercote Due to `enum PropertyDeclaration`, we must `impl MallocSizeOf for TemplateAreasArc` so I cannot only `impl MallocConditionalSizeOf` for it.
How do you think about it?

cc @emilio

Reply:
Is `#[derive(MallocSizeOf)]` involved? It works well for tree-like structures, but less well for graph-like structures. One possibility is to change the relevant `MallocSizeOf` impl from a derived one to a manually written one that does something appropriate (and different) for the Arc. Another possibility is to keep the `derive` impl and use the `ignore_malloc_size_of` attribute on the relevant `Arc` field.

Reply:
Yes, this code block is involved `#[derive(MallocSizeOf)]`.

https://github.com/servo/servo/blob/0fa605d2439c4f640b7f180c18d768a7a14e7f1e/components/style/properties/properties.mako.rs#L1404-L1408

I think adding another checking into mako for only this case might not be good.
I'll try to use `ignore_malloc_size_of` attribute for it, thanks.

Reply:
I tried

```rust
pub struct TemplateAreasArc(#[ignore_malloc_size_of = "Arc"] pub Arc<TemplateAreas>);
```

However, I got an error of

```
error: The attribute `ignore_malloc_size_of` is currently unknown to the compiler and may have meaning added to it in the future
   --> components/style/values/specified/position.rs:620:29
    |
620 | pub struct TemplateAreasArc(#[ignore_malloc_size_of = "Arc"] pub Arc<TemplateAreas>);
    |                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error
```

Should I import anything for it or did I miss anything for using this annotation, `#[ignore_malloc_size_of]`?